polars/polars-lazy/src/frame/mod.rs

```diff
@@ -907,7 +887,12 @@ impl LazyFrame {
     ///         .left_join(other, col("foo"), col("bar"))
     /// }
     /// ```
-    pub fn left_join<E: Into<Expr>>(self, other: LazyFrame, left_on: E, right_on: E) -> LazyFrame {
+    pub fn left_join<L: Into<Expr>, R: Into<Expr>>(
```

Comment:
this is a drive-by - before, lhs and rhs had to be of the same type. However, if we want to support `Into<Expr>`, we should allow independent types to be passed to each side.